tools/cldr-apps/js/src/views/ForumForm.vue

```diff
@@ -0,0 +1,140 @@
+<script setup>
+import { onMounted, reactive, ref } from "vue";
+
+import * as cldrForum from "../esm/cldrForum.mjs";
+import * as cldrForumType from "../esm/cldrForumType.mjs";
+import * as cldrText from "../esm/cldrText.mjs";
+
+const DEBUG_FORUM_FORM = false;
+
+const formRef = ref();
+
+const props = defineProps({
+  pi: Object,
+  reminder: String,
+});
+
+const formState = reactive({
+  body: cldrForum.prefillPostText(props.pi),
+});
+
+const rules = {
+  body: [
+    {
+      validator: validateBody,
+      trigger: "change",
+    },
+  ],
+};
+
+const emit = defineEmits(["submit-or-cancel"]);
+
+const textAreaToFocus = ref(null);
+
+function focusInput() {
+  if (textAreaToFocus.value) {
+    textAreaToFocus.value.focus();
+  }
+}
+
+onMounted(focusInput);
+
+async function validateBody(_rule, bodyText) {
+  if (!bodyText.trim()) {
+    return Promise.reject("Please enter a message body");
+  } else if (props.pi.postType === cldrForumType.REQUEST) {
+    const prefill = cldrText.sub("forum_prefill_request", [props.pi.value]);
+    if (bodyText.trim() === prefill.trim()) {
+      return Promise.reject("Please edit the message body");
+    }
+  }
+  return Promise.resolve();
+}
+
+function onCancel() {
+  emit("submit-or-cancel", null);
+}
+
+function onSubmit() {
+  formRef.value
+    .validate()
+    .then(() => {
+      emit("submit-or-cancel", formState);
+    })
+    .catch((error) => {
+      if (DEBUG_FORUM_FORM) {
+        console.log("onSubmit validation error", error);
+      }
+    });
+}
+
+function formatParent(parentPost) {
+  const div = cldrForum.parseContent([parentPost], "parent");
+  return new XMLSerializer().serializeToString(div);
```

Comment:
This method takes DOM elements generated by legacy code (cldrForum.parseContent) and turns them into html using v-html. In the short term this is easy and achieves consistency with the legacy formatting. Eventually of course there should be a Vue component for rendering completed forum posts.